### ISSUE TITLE
Version 1.1.1 makes breaking changes to Foreign.Lua.pushList

### ISSUE THREAD

**Issue by phlummox** [hslua/hslua#83]
As per the Haddock documentation: the signature for pushList has changed [from](https://hackage.haskell.org/package/hslua-1.1.0/docs/Foreign-Lua.html#v:pushList)

```haskell
pushList :: Pushable a => [a] -> Lua ()
```

[to](https://hackage.haskell.org/package/hslua-1.1.1/docs/Foreign-Lua.html#v:pushList):

```haskell
pushList :: (a -> Lua ()) -> [a] -> Lua ()
```

and code using version 1.1.0 will therefore not compile with 1.1.1. An alternative might have been just to deprecate the old function, but given that it *has* changed, this should be documented in the Haddocks.

(It's also typical for breaking changes to be reflected by a major version bump, as per the Haskell [Package Versioning Policy](https://pvp.haskell.org/#decision-tree) -- if that policy is applied, that would make this version 1.2.0, not 1.1.1.)

### Example of breakage: 

https://github.com/hslua/hslua-aeson/issues/2

**Comment by tarleb** [comment]
Yikes! That change should not have been in there and slipped through.
Not good, thanks for catching it.

I'll release a proper 1.1.2 version soon to unbreak the PVP, and
re-release the breaking changes as 1.2.0.

**Comment by tarleb** [comment]
I reverted the signature change of `pushList` and released version 1.1.2. Additionally, version 1.1.1 has been marked as deprecated on Hackage.

**Comment by phlummox** [comment]
Cheers! Many thanks :)
